src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -247,24 +327,79 @@ public void Dispose()
         /// </summary>
         /// <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
         protected virtual void Dispose(bool disposing)
+        {
```

Comment:
I think this Dispose method should follow the standard dispose pattern:

    protected virtual void Dispose(bool disposing)
    {
        if (disposing)
        {
            if (!_disposed)
            {
                try
                {
                    using (var source = new CancellationTokenSource(TimeSpan.FromSeconds(10)))
                    {
                        CloseCursor(source.Token);
                    }
                }
                catch
                {
                    // ignore exceptions
                }
                if (_channelSource != null)
                {
                    _channelSource.Dispose();
                }
                _disposed = true;
            }
        }
    }